crates/primitives-traits/src/account.rs

```diff
@@ -66,9 +66,14 @@ impl Compact for Bytecode {
     where
         B: bytes::BufMut + AsMut<[u8]>,
     {
-        let bytecode = &self.0.bytecode()[..];
+        let bytecode = match &self.0 {
+            RevmBytecode::LegacyRaw(bytes) => bytes,
+            RevmBytecode::LegacyAnalyzed(analyzed) => analyzed.bytecode(),
+            RevmBytecode::Eof(eof) => eof.raw(),
+            RevmBytecode::Eip7702(eip7702) => eip7702.raw(),
+        };
         buf.put_u32(bytecode.len() as u32);
-        buf.put_slice(bytecode);
+        buf.put_slice(bytecode.as_ref());
```

Comment:
fyi @onbjerg

Reply:
looks correct

Reply:
this is required for the codec I believe